rdt-app/src/main/java/io/ona/rdt_app/fragment/RDTJsonFormFragment.java

```diff
@@ -44,13 +45,20 @@ public void onClick(View v) {
         rootView.findViewById(com.vijay.jsonwizard.R.id.next_button).setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                boolean isSaved = save(false);
-                if (isSaved) {
-                    try {
-                        jsonFormUtils.launchForm(getActivity(), RDT_TEST_FORM);
-                    } catch (JSONException e) {
-                        Log.e(TAG, e.getStackTrace().toString());
+                Object isSubmit = v.getTag(R.id.submit);
+                if (isSubmit != null && Boolean.valueOf(isSubmit.toString())) {
+                    boolean isSaved = save(false);
+                    if (isSaved) {
+                        try {
+                            jsonFormUtils.launchForm(getActivity(), RDT_TEST_FORM);
+                        } catch (JSONException e) {
+                            Log.e(TAG, e.getStackTrace().toString());
+                        }
+                    } else {
```

Comment:
Why do we need this next here?

It's already part of the outer if-else block.

The next button is either of type submit or simply type next.